extensions/wikia/PortableInfobox/services/Parser/Nodes/NodeImage.php

```diff
@@ -10,43 +10,67 @@ class NodeImage extends Node {
 	const CAPTION_TAG_NAME = 'caption';
 	const MEDIA_TYPE_VIDEO = 'VIDEO';
 
+	private function getImage($key, $caption, $alt) {
+		$title = $this->getImageAsTitleObject( $key );
+		$file = $this->getFilefromTitle( $title );
+		if ( $title instanceof \Title ) {
+			$this->getExternalParser()->addImage( $title->getDBkey() );
+		}
+		$ref = null;
+		wfRunHooks( 'PortableInfoboxNodeImage::getData', [ $title, &$ref, $caption ] );
+		$item = [
+			'url' => $this->resolveImageUrl( $file ),
+			'name' => ( $title ) ? $title->getText() : '',
+			'key' => ( $title ) ? $title->getDBKey() : '',
+			'alt' => $alt,
+			'caption' => $caption,
+			'ref' => $ref
+		];
+		if ( $this->isVideo( $file ) ) {
+			$item = $this->videoDataDecorator( $item, $file );
+		}
+		return $item;
+	}
+
 	public function getData() {
 		if ( !isset( $this->data ) ) {
 			$imageData = $this->getRawValueWithDefault( $this->xmlNode );
-
-			if( is_string($imageData) && PortableInfoboxDataBag::getInstance()->getGallery($imageData)) {
-				$imageData = PortableInfoboxDataBag::getInstance()->getGallery($imageData);
-			}
-
-			$title = $this->getImageAsTitleObject( $imageData );
-			$file = $this->getFilefromTitle( $title );
-			if ( $title instanceof \Title ) {
-				$this->getExternalParser()->addImage( $title->getDBkey() );
-			}
-			$ref = null;
-			$alt = $this->getValueWithDefault( $this->xmlNode->{self::ALT_TAG_NAME} );
-			$caption = $this->getValueWithDefault( $this->xmlNode->{self::CAPTION_TAG_NAME} );
-
-			$this->data = [
-				'url' => $this->resolveImageUrl( $file ),
-				'name' => ( $title ) ? $title->getText() : '',
-				'key' => ( $title ) ? $title->getDBKey() : '',
-				'alt' => $alt,
-				'caption' => $caption
-			];
-
-			if ( $this->isVideo( $file ) ) {
-				$this->data = $this->videoDataDecorator( $this->data, $file );
+			$imageDataLower = strtolower($imageData);
+
+			if ( strpos( $imageDataLower, '-tabber-' ) !== false || strpos( $imageDataLower, '-gallery-' ) !== false ) {
+				$this->data = [];
+				$parsed = $this->getExternalParser()->parseRecursive( $imageData );
+
+				$galleryImages = $this->getGalleryItems( $imageData );
+				$tabberImages = $this->getTabberItems( $parsed );
+
+				$images = $galleryImages + $tabberImages;
+				for( $i = 0; $i < count( $images ); $i++ ) {
+					$this->data[] = $this->getImage(
+						$images[$i]['key'],
+						$images[$i]['caption'],
+						$images[$i]['alt']
+					);
+				}
+			} else {
+				$this->data = array( $this->getImage(
```

Comment:
There was no feedback about changing NodeImage to return array of images instead of just one image so I'm assuming all interested people are ok with it.